test/client/api.js

```diff
@@ -671,6 +671,18 @@ module.exports = config => {
     )
   }
 
+  ClientApi.prototype.consumeSigninCode = function (code, metricsContext) {
+    return this.doRequest(
+      'POST',
+      `${this.baseURL}/signinCodes/consume`,
+      null,
+      {
+        code: code.toString('base64').replace(/\+/g, '-').replace(/\//g, '_').replace(/=/g, ''),
```

Comment:
Not sure I understand the question.

The conversion here is because that is what the server expects to receive, but maybe you're really asking why the server expects a base64-encoded string or why it is a URL-safe base64 encoded string? That's just so the same code character-for-character is consumed as is sent out in the SMS message, clients shouldn't need to know that it is a base64-encoded binary value, just that it is some characters.

Does that answer your question?